includes/class-dispatcher.php

```diff
@@ -126,9 +126,18 @@ public static function process_outbox( $id ) {
 	 * @return array|void The next batch of followers to process, or void if done.
 	 */
 	public static function send_to_followers( $outbox_item_id, $batch_size = ACTIVITYPUB_OUTBOX_PROCESSING_BATCH_SIZE, $offset = 0 ) {
-		$json    = Outbox::get_activity( $outbox_item_id )->to_json();
-		$actor   = Outbox::get_actor( \get_post( $outbox_item_id ) );
-		$inboxes = Followers::get_inboxes_for_activity( $json, $actor->get__id(), $batch_size, $offset );
+		$activity = Outbox::get_activity( $outbox_item_id );
+		$json     = $activity->to_json();
+		$actor    = Outbox::get_actor( \get_post( $outbox_item_id ) );
+		$is_move  = $activity->get_type() === 'Move';
+		$batch    = $is_move ? -1 : $batch_size; // -1 for Move activities to get all inboxes at once before deleting below.
+		$inboxes  = Followers::get_inboxes_for_activity( $json, $actor->get__id(), $batch, $offset );
+
+		if ( $is_move && ! empty( $inboxes ) ) {
+			// We are sending a `Move` which means that we need to remove all Followers from the old Actor.
+			// The new Actor will receive the `Follow` activity from any Followers that choose to Follow it.
+			Followers::remove_all_followers( $actor->get__id() );
+		}
```

Comment:
Hm, I'm not sure this feels like the right place for it. I guess there's a race condition we're trying to avoid with this, where waiting until the batches are complete would be too late?

Reply:
What if we added a post meta for Followers that keeps track of the actor id and we differentiate `Undo` requests based on that? Add an upgrade routine that sets it for all followers?

Reply:
I would not remove them at all!

Reply:
My understanding is that follower instance will start sending Undo Follow requests to the old ID and Follow requests to the new ID (which it probably expects to be accepted).

The expectation that the follower list for the old ID is empty [is part of the draft spec](https://swicg.github.io/activitypub-data-portability/#results-0).

Reply:
Fixed in [a733aca](https://github.com/Automattic/wordpress-activitypub/pull/1530/commits/a733aca3e0231f08e2bebbb2188f39c632b6335b)